### DISCUSSION TITLE
Using Session Request in Query Builder

### DISCUSSION THREAD

**Discussion by nubinter** [rappasoft/laravel-livewire-tables#784] [Q&A]
How can i use session request in the query builder?
if i try adding `Request $request` inside Builder() methode, i will get the error 
Declaration of App\Http\Livewire\Cuti\Class::builder(Request $request): Illuminate\Database\Eloquent\Builder must be compatible with Rappasoft\LaravelLivewireTables\DataTableComponent::builder(): Illuminate\Database\Eloquent\Builder

so is there any other way to access laravel session? because i need the data to use in query builder.

**Comment by inmanturbo** [comment]
@nubinter yes, you don't have to rely on passing the request down through `DI`. You can always access the session from anywhere in laravel using the `session()` helper https://laravel.com/docs/9.x/helpers#method-session
